src/effects/effect_1.c

```diff
@@ -1,16 +1,185 @@
+f32 fx_1_partYaws[10] = { 0.0f, 234.0f, 468.0f, 702.0f, 936.0f, 1260.0f, 1404.0f, 1638.0f, 1902.0f, 1976.0f };
+
+s32 fx_1_dlists[7] = { 0x09000FA0, 0x09001060, 0x09001120, 0x090011E0, 0x090012A0, 0x09001360, 0x09001420 };
+
+void fx_1_main(f32 x, f32 y, f32 z) {
+    EffectBlueprint bp;
+    EffectInstance* effect;
+    Effect1* effectData;
+    f32 theta;
+    f32 temp_f20;
+    f32 cosTheta;
+    f32 sinTheta;
+    s32 numParts = 10;
+    s32 i;
+
+    bp.unk_00 = 0;
+    bp.init = fx_1_init;
+    bp.update = fx_1_update;
+    bp.renderWorld = fx_1_render;
+    bp.unk_14 = NULL;
+    bp.effectIndex = 1;
+
+    effect = shim_create_effect_instance(&bp);
+    effect->numParts = numParts;
+
+    effectData = shim_general_heap_malloc(effect->numParts * sizeof(Effect1));
+    effect->data = effectData;
+
+    ASSERT(effectData != NULL);
+
+    for (i = 0; i < effect->numParts; i++, effectData++) {
+        effectData->unk_00 = 0;
+        effectData->unk_02 = effectData->unk_04 = func_E0200000(6) + 0x10;
+        effectData->unk_06 = 4;
+        effectData->unk_08 = 0;
+        effectData->x = x;
+        effectData->y = y;
+        effectData->z = z;
+        effectData->partX = 0;
+        effectData->partY = 0;
+
+        theta = D_E0002760[i] + 90.0f;
+        temp_f20 = 2.0 * D_E0002788[i];
+        sinTheta = shim_sin_deg(theta);
+        cosTheta = shim_cos_deg(theta);
+
+        effectData->unk_20 = temp_f20 * sinTheta;
+        effectData->unk_24 = -temp_f20 * cosTheta;
+    }
+}
+
+void fx_1_init(EffectInstance* effect) {
 }
 
-INCLUDE_ASM(s32, "effects/effect_1", func_E00021BC);
+void fx_1_update(EffectInstance* effect) {
+    Effect1* effectData = effect->data;
+    s32 cond = FALSE;
+    s32 i;
+
+    for (i = 0; i < effect->numParts; i++, effectData++) {
+        if (effectData->unk_02 >= 0) {
+            effectData->unk_08--;
+
+            if (effectData->unk_08 < 0) {
+                effectData->partX += effectData->unk_20;
+                effectData->partY += effectData->unk_24;
+                effectData->unk_00++;
+                effectData->unk_02--;
+                cond = TRUE;
+
+                if (effectData->unk_02 >= 0) {
+                    effectData->unk_06--;
+                    if (effectData->unk_06 < 0) {
+                        effectData->unk_20 /= 1.25;
+                        effectData->unk_24 /= 1.25;
+                    }
+                }
+            }
+        }
+    }
+
+    if (!cond) {
+        shim_remove_effect(effect);
+    }
+}
 
-INCLUDE_ASM(s32, "effects/effect_1", func_E00022BC);
+void fx_1_render(EffectInstance* effect) {
+    RenderTask renderTask;
+    RenderTask* retTask;
 
-INCLUDE_ASM(s32, "effects/effect_1", func_E0002300);
+    renderTask.appendGfx = fx_1_appendGfx;
+    renderTask.appendGfxArg = effect;
+    renderTask.distance = 0;
+    renderTask.renderMode = RENDER_MODE_SHADOW | RENDER_MODE_ALPHATEST; // RENDER_MODE_2D?
```

Comment:
Yeah I don't think these work like flags

Reply:
It's weird though because it gets `or`ed with 2 later on. We should decomp the function that reads this